deploy/doc.Dockerfile

```diff
@@ -3,15 +3,19 @@
 # https://hub.docker.com/_/golang
 FROM golang:1.13 as builder
 
-WORKDIR app/
+WORKDIR /go/app/
 
-# Copy internal libraries.
-COPY . .
+# Copy and cache dependencies
+COPY ../go.mod .
+COPY ../go.sum .
```

Comment:
This `Dockerfile` typically gets invoked from root via the `Makefile` so I think we can keep this as is (the proposed change will not work with `make build-doc` or `make build-gitter`.

Reply:
`make build-doc` and `make build-gitter` both work for me. Did you have any problems?
Up to you, I just feel these changes make it easier to iterate during development.

Analyzing execution of `make build-doc`...
Based on the `main` branch, the 1st execution took me 78s. Another execution on top of that without any changes was quick, 1.8s. Then I just added a comment to `deploy/doc.yaml` (`echo "\n# test" >> deploy/doc.yaml`), increased timing to 56s.

Now with my proposed changes, the 2 first executions were very similar. But the 3rd execution, which has an updated file, took just 13s vs 56s.

Again, I'm not working often with the repo to say if that's important. But when I had to build the image several times, that saved me quite some time

Reply:
Just for reference, running these before `go mod download` is suggested in [golang's Dockerhub](https://hub.docker.com/_/golang) under "Start a Go instance in your app"

Reply:
@tyron yep I am on board with copying just the `go.mod` / `go.sum` files in for `go mod download` to ensure that we can cache and re-use the intermediate image 👍🏻 The `Makefile` is in the root of the directory though, as is the build context, so your relative paths to `../go.mod` and `../go.sum` should just be `go.mod` and `go.sum`.

Reply:
FWIW we can also mount the go build cache as well if you want to speed it up even more :)

Reply:
Hey, so I tested a bit more because I was curious on how my setup was working. Seems that running in the root as `docker build . -f deploy/doc.Dockerfile` (Makefile) or inside deploy folder as just `docker build . -f doc.Dockerfile` has the same results -- and that always finds the correct files. 

I believe the notation in the Dockerfile is not based on your current pwd, yet relative to Dockerfile itself. So since `go.{mod,sum}` is always on the parent folder of `deploy`, writing as `../go.{mod,sum}` always returns the proper location.

Reply:
Ok, now I don't understand anything. Both `COPY ../go.mod .` and `COPY go.mod .` seem to work.